index.js

```diff
@@ -130,6 +132,106 @@ module.exports = {
+   * 
+   * @param {!object} jso - The JSON-RPC call to make.
+   * @param {?string} preferredTransport - IPC, WS, HTTP, SYNC or null.  If not null, will only use the transport specified.  If null, will use the best available.
+   * @param {?function(?Error, ?object):void} callback - Called when a response to the request is received.  May only be null if preferredTransport is SYNC.
+   * @returns {void|?Error|?object} - Returns the error or result if the operation is synchronous.
+   */
+  submitRequestToBlockchain: function (jso, preferredTransport, callback) {
+    var syncErrorOrResult;
+    if (preferredTransport === "SYNC") callback = function (error, result) { return syncErrorOrResult = error || result };
+
+    if (typeof callback !== "function") throw new Error("callback must be a function");
+    if (typeof preferredTransport !== "string" && preferredTransport !== null) return callback(new Error("preferredTransport must be null or a string")); 
+    if (typeof jso !== "object") return callback(new Error("jso must be an object"));
+    if (typeof jso.id !== "number") return callback(new Error("jso.id must be a number"));
+
+    // FIXME: return types shouldn't be embedded into the RPC JSO
+    var expectedReturnTypes = this.strip(jso);
+    this.internalState.outstandingRequests[jso.id] = {
+      jso: jso,
+      expectedReturnTypes: expectedReturnTypes,
+      callback: callback
+    };
+
+    this.internalState.transporter.blockchainRpc(jso, preferredTransport);
+
+    if (preferredTransport === "SYNC") {
+      if (typeof this.internalState.outstandingRequests[jso.id] !== "undefined") return new Error("SYNC request didn't receive messageHandler call before returning.");
+      return syncErrorOrResult;
+    }
+  },
+
+  /**
+   * Used internally.  Processes a response from the blockchain by looking up the associated callback and calling it.
+   */
+  blockchainResponseHandler: function (error, responseJso) {
+    if (error !== null) return this.configuration.errorHandler(error);
+    if (typeof responseJso !== "object") return this.configuration.errorHandler(new ErrorWithData("Unexpectedly received a message from the transport that was not an object.", responseJso));
+    if (typeof responseJso.id !== "number") return this.configuration.errorHandler(new ErrorWithData("Received a message from the blockchain that didn't have a valid id.", responseJso));
+
+    var outstandingRequest = this.internalState.outstandingRequests[responseJso.id];
+    delete this.internalState.outstandingRequests[responseJso.id];
+    if (typeof outstandingRequest !== "object") return this.configuration.errorHandler(new ErrorWithData("Unable to locate original request for blockchain response.", responseJso));
+    
+    // FIXME: outstandingRequest.callback should be function(Error,object) not function(Error|object)
+    this.parse(responseJso, outstandingRequest.expectedReturnTypes, outstandingRequest.callback);
```

Comment:
@tinybike You indicated that callbacks should all be of the form `function (error, result)` but the callbacks in ethrpc appear to all be `function (errorOrResult)`.  For now I went ahead with keeping the signature the same as I didn't want to do that big of a breaking change, but I did want to call out that fixing it shouldn't be too hard in `ethrpc` with this change.  It will require every call site into `ethrpc` being touched though.  :/